### PR TITLE
feat: make Token pausable

### PR DESCRIPTION
**Pull request by bowd**
### Description

Based on recent token-related conversations we've decided we want more flexibility when it comes to the time between when we launch governance and give utility to the token, and the time when the token is transferable, i.e. tradable.
Most recent token launches have followed this approach, most notably Safe.
These changes make the token ownable and pausable. 

However, even when paused the token can still be transferred by:
- The owner
- The locking contract - it can be locked and unlocked, thus the airgrab can run while paused
- The emissions contract - it can be minted into existence while paused

### Other changes

N/A

### Tested

- Tests have been added for the paused scenario
 
### Backwards compatibility

N/A

### Documentation

N/A

### CODE REVIEW

File: contracts/governance/MentoToken.sol [hunk 1]
```diff
@@ -69,4 +90,27 @@ contract MentoToken is ERC20Burnable {
...
+    address to,
+    uint256 amount
+  ) internal virtual override {
+    super._beforeTokenTransfer(from, to, amount);
+
+    require(to != address(this), "MentoToken: cannot transfer tokens to token contract");
+    // Token transfers are only possible if the contract is not paused
+    // OR if triggered by the owner of the contract
+    // OR if triggered by the locking contract
+    // OR if triggered by the emission contract
```
Comment (chapati23) [inline contracts/governance/MentoToken.sol:110]: appreciate the extensive comments

File: test/governance/MentoToken.t.sol [hunk 2]
```diff
@@ -177,4 +200,60 @@ contract MentoTokenTest is TestSetup {
...
+  function test_transferFrom_whenPaused_calledByLocking_shouldWork() public {
+    uint256 amount = 10e18;
+    deal(address(mentoToken), alice, amount);
+    vm.prank(alice);
+    mentoToken.approve(locking, amount);
+
+    vm.prank(locking);
+    mentoToken.transferFrom(alice, bob, amount);
+    assertEq(mentoToken.balanceOf(bob), amount);
+  }
```
Comment (chapati23) [inline test/governance/MentoToken.t.sol:258]: should we also check that Emissions work while contract is paused?

**Review by chapati23 — APPROVED** [review]
looks solid, just one question on testing transfer from Emissions while paused

**Review by philbow61 — APPROVED** [review]
LGTM

File: contracts/governance/MentoToken.sol [hunk 3]
```diff
@@ -2,13 +2,18 @@
...
+import { Pausable } from "openzeppelin-contracts-next/contracts/security/Pausable.sol";
 
 /**
  * @title Mento Token
  * @author Mento Labs
  * @notice This contract represents the Mento Protocol Token which is a Burnable ERC20 token.
  */
-contract MentoToken is ERC20Burnable {
+contract MentoToken is Ownable, Pausable, ERC20Burnable {
+  /// @notice The address of the emission contract that has the capability to emit new tokens.
```
Comment (nvtaveras) [inline contracts/governance/MentoToken.sol:14]: update?

File: contracts/governance/MentoToken.sol [hunk 4]
```diff
@@ -50,9 +61,19 @@ contract MentoToken is ERC20Burnable {
...
+  }
+
+  /**
+   * @notice Unpauses all token transfers.
+   * @dev See {Pausable-_unpause}
+   * Requirements: caller must be the owner
+   */
+  function unpause() public virtual onlyOwner {
+    require(paused(), "MentoToken: token is not paused");
+    _unpause();
```
Comment (nvtaveras) [inline contracts/governance/MentoToken.sol:76]: couldn't find tests for this function?

**Review by nvtaveras — APPROVED** [review]

**Review by nvtaveras — CHANGES_REQUESTED** [review]
a few small things

File: contracts/governance/MentoToken.sol [hunk 5]
```diff
@@ -2,14 +2,21 @@
...
 
 /**
  * @title Mento Token
  * @author Mento Labs
  * @notice This contract represents the Mento Protocol Token which is a Burnable ERC20 token.
  */
-contract MentoToken is ERC20Burnable {
+contract MentoToken is Ownable, Pausable, ERC20Burnable {
+  /// @notice The address of the lockin contract that has the capability to transfer tokens
+  /// even when the contarct is paused.
```
Comment (nvtaveras) [inline contracts/governance/MentoToken.sol:15]:
```suggestion
  /// @notice The address of the locking contract that has the capability to transfer tokens
  /// even when the contract is paused.
```

File: contracts/governance/MentoToken.sol [hunk 6]
```diff
@@ -2,14 +2,21 @@
...
  * @author Mento Labs
  * @notice This contract represents the Mento Protocol Token which is a Burnable ERC20 token.
  */
-contract MentoToken is ERC20Burnable {
+contract MentoToken is Ownable, Pausable, ERC20Burnable {
+  /// @notice The address of the locking contract that has the capability to transfer tokens
+  /// even when the contract is paused.
+  address public immutable locking;
+
   /// @notice The address of the emission contract that has the capability to emit new tokens.
```
Comment (nvtaveras) [inline contracts/governance/MentoToken.sol:18]:
```suggestion
  /// @notice The address of the emission contract that has the capability to emit new tokens
```

**Review by nvtaveras — APPROVED** [review]

**Pull request merged**